service/internal/telemetrylogs/logger.go

```diff
@@ -0,0 +1,139 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetrylogs
+
+import (
+	"flag"
+	"fmt"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+const (
+	logLevelCfg   = "log-level"
+	logProfileCfg = "log-profile"
+	logFormatCfg  = "log-format"
+)
+
+var (
+	// Command line pointer to logger level flag configuration.
+	loggerLevelPtr   *string
+	loggerProfilePtr *string
+	loggerFormatPtr  *string
+)
+
+// Flags adds flags related to service telemetry logs to the given flagset.
+// Deprecated: keep this flag for preventing the breaking change. Use `service::telemetry::logs` in config instead.
+func Flags(flags *flag.FlagSet) {
+	loggerLevelPtr = flags.String(logLevelCfg, "deprecated", "Deprecated, DO NOT USE")
+
+	loggerProfilePtr = flags.String(logProfileCfg, "deprecated", "Deprecated, DO NOT USE")
+
+	// Note: we use "console" by default for more human-friendly mode of logging (tab delimited, formatted timestamps).
+	loggerFormatPtr = flags.String(logFormatCfg, "deprecated", "Deprecated, DO NOT USE")
+}
+
+func NewLogger(cfg config.ServiceTelemetryLogs, options []zap.Option) (*zap.Logger, error) {
+	// Copied from NewProductionConfig.
+	zapCfg := &zap.Config{
+		Level:       zap.NewAtomicLevelAt(zap.InfoLevel),
+		Development: false,
+		Sampling: &zap.SamplingConfig{
+			Initial:    100,
+			Thereafter: 100,
+		},
+		Encoding:         "json",
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		OutputPaths:      []string{"stderr"},
```

Comment:
stdout?

Reply:
As the comment says, it is copied and that's how it is defined there.
https://github.com/uber-go/zap/blob/master/config.go#L120